volmdlr/faces.py

```diff
@@ -5060,6 +5067,35 @@ def copy(self, deep=True, memo=None):
         return self.__class__(self.surface3d.copy(deep, memo), self.surface2d.copy(),
                               self.name)
 
+    def face_inside(self, face2):
+        """
+        Verifies if a planeface is inside another planeface.
+
+        It returns True if face2 is inside or False if the opposite.
+        """
+        if not type(self) is type(face2):
+            return False
+        if self.surface3d.is_coincident(face2.surface3d):
+            self_contour2d = self.outer_contour3d.to_2d(
+                self.surface3d.frame.origin, self.surface3d.frame.u, self.surface3d.frame.v)
+            face2_contour2d = face2.outer_contour3d.to_2d(
+                self.surface3d.frame.origin, self.surface3d.frame.u, self.surface3d.frame.v)
+            if self_contour2d.is_inside(face2_contour2d):
+                return True
+        return False
+
+    def edge_intersections(self, edge):
+        intersections = []
+        method_name = f'{edge.__class__.__name__.lower()[:-2]}_intersections'
+        if hasattr(self, method_name):
+            intersections = getattr(self, method_name)(edge)
+        if not intersections:
+            for point in [edge.start, edge.end]:
+                if self.point_belongs(point):
+                    if point not in intersections:
+                        intersections.append(point)
+        return intersections
```

Comment:
This is a generalised method created to calculate the intersections of different edges with a face3D